examples/coordinates/plot_mars-coordinate-frame.py

```diff
@@ -0,0 +1,154 @@
+    BaseBodycentricRepresentation,
+    BaseGeodeticRepresentation,
+)
+
+##############################################################################
+# The first step is to create a new class, and make it a subclass of
+# `~astropy.coordinates.BaseGeodeticRepresentation`.
+# Geodetic latitudes are used and longitudes span from 0 to 360 degrees east positive
+# It represent a best fit of the Mars spheroid to the martian geoid (areoid):
+
+class MarsBestFitAeroid(BaseGeodeticRepresentation):
+    """
+    A Spheroidal representation of Mars that minimized deviations with respect to the
+    areoid following
+        Ardalan A. A, R. Karimi, and E. W. Grafarend (2010)
+        https://doi.org/10.1007/s11038-009-9342-7
+    """
+
+    _equatorial_radius = 3395.4280 * u.km
+    _flattening = 0.5227617843759314 * u.percent
+
+#####################################################################################
+# Now let's define a new geodetic representation obtained from MarsBestFitAeroid but
+# described by planetocentric latitudes.
+
+class MarsBestFitOcentricAeroid(BaseBodycentricRepresentation):
+    """
+    A Spheroidal planetocentric representation of Mars that minimized deviations with
+    respect to the areoid following
+        Ardalan A. A, R. Karimi, and E. W. Grafarend (2010)
+        https://doi.org/10.1007/s11038-009-9342-7
+    """
+
+    _equatorial_radius = 3395.4280 * u.km
+    _flattening = 0.5227617843759314 * u.percent
+
+#############################################################################
+# As a comparison we define a new spherical frame representation, we could
+# have based it on `~astropy.coordinates.BaseBodycentricRepresentation` too.
+
+class MarsSphere(BaseGeodeticRepresentation):
+    """
+    A Spherical representation of Mars
+    """
+
+    _equatorial_radius = 3395.4280 * u.km
+    _flattening = 0.0 * u.percent
+
+#############################################################################
+# The new planetary body-fixed reference system will be described using the
+# previous defined representations.
+
+class MarsCoordinateFrame(BaseCoordinateFrame):
+    """
+    A reference system for Mars.
+    """
+
+    name = "Mars"
+
+
+#############################################################################
+# Now we plot the differences between each component of the cartesian
+# representation with respect to the spherical model, assuming the point on the
+# surface of the body (``height = 0``)
+
+mars_sphere = MarsCoordinateFrame(
+    lon=np.linspace(0, 360, 128)*u.deg,
+    lat=np.linspace(-90, 90, 128)*u.deg,
+    representation_type=MarsSphere)
+mars = MarsCoordinateFrame(
+    lon=np.linspace(0, 360, 128)*u.deg,
+    lat=np.linspace(-90, 90, 128)*u.deg,
+    representation_type=MarsBestFitAeroid)
+mars_ocentric = MarsCoordinateFrame(
+    lon=np.linspace(0, 360, 128)*u.deg,
+    lat=np.linspace(-90, 90, 128)*u.deg,
+    representation_type=MarsBestFitOcentricAeroid)
```

Comment:
Just to be sure, is this plot the one you expected?
![image](https://github.com/astropy/astropy/assets/2789820/53acf0d6-eb03-4ee7-8f94-d22b255f647e)

Reply:
> is this plot the one you expected?

Yes, I checked the trends with lat and long and it seems to me that they are correct.... their readability could probably be improved...

Reply:
OK. I wasn't quite sure how to read the figure, but after a bit more staring think I've got it. Not quite sure how to improve it, so maybe fine to leave it as is.